labelCloud/control/controller.py

```diff
@@ -273,46 +275,100 @@ def key_press_event(self, a0: QtGui.QKeyEvent) -> None:
+        elif a0.key() in [Keys.Key_S]:
             # move forward
             self.bbox_controller.translate_along_y(forward=True)
-        elif (a0.key() == QtCore.Qt.Key_A) or (a0.key() == QtCore.Qt.Key_Left):
+        elif a0.key() in [Keys.Key_A]:
             # move left
             self.bbox_controller.translate_along_x(left=True)
-        elif (a0.key() == QtCore.Qt.Key_D) or (a0.key() == QtCore.Qt.Key_Right):
+        elif a0.key() in [Keys.Key_D]:
             # move right
             self.bbox_controller.translate_along_x()
-        elif (a0.key() == QtCore.Qt.Key_Q) or (a0.key() == QtCore.Qt.Key_PageUp):
+        elif a0.key() in [Keys.Key_Q]:
             # move up
             self.bbox_controller.translate_along_z()
-        elif (a0.key() == QtCore.Qt.Key_E) or (a0.key() == QtCore.Qt.Key_PageDown):
+        elif a0.key() in [Keys.Key_E]:
             # move down
             self.bbox_controller.translate_along_z(down=True)
+        elif a0.key() in [Keys.Key_R, Keys.Key_Left]:
+            # load previous sample
+            self.prev_pcd()
+        elif a0.key() in [Keys.Key_F, Keys.Key_Right]:
+            # load next sample
+            self.next_pcd()
+        elif a0.key() in [Keys.Key_T, Keys.Key_Up]:
+            # select previous bbox
+            new_bbox_id = self.bbox_controller.active_bbox_id - 1
+            self.bbox_controller.set_active_bbox(new_bbox_id)
+        elif a0.key() in [Keys.Key_G, Keys.Key_Down]:
+            # select previous bbox
+            new_bbox_id = self.bbox_controller.active_bbox_id - 1
+            self.bbox_controller.set_active_bbox(new_bbox_id)
+        elif a0.key() in [Keys.Key_Comma]:
+            # change bbox class to previous available class
+            classes = [cls_cfg.name for cls_cfg in LabelConfig().classes]
+            current_class = self.bbox_controller.get_active_bbox().get_classname()
+            curr_idx = classes.index(current_class)
+            prev_idx = curr_idx - 1
+            prev_idx = prev_idx if prev_idx >= 0 else len(classes) - 1
+            new_class = classes[prev_idx]
+            self.bbox_controller.get_active_bbox().set_classname(new_class)
+            self.bbox_controller.update_all()  # updates UI in SelectBox
+        elif a0.key() in [Keys.Key_Period]:
+            # change bbox class to next available class
+            classes = [cls_cfg.name for cls_cfg in LabelConfig().classes]
+            current_class = self.bbox_controller.get_active_bbox().get_classname()
+            curr_idx = classes.index(current_class)
+            next_idx = curr_idx + 1
+            next_idx = next_idx if next_idx < len(classes) else 0
+            new_class = classes[next_idx]
+            self.bbox_controller.get_active_bbox().set_classname(new_class)
+            self.bbox_controller.update_all()  # updates UI in SelectBox
+
+        # select bboxes with digit keys
+        elif a0.key() in [Keys.Key_1]:
```

Comment:
I wonder if we could extract the int value from the key object but didn't find sth. with a quick search ...

Reply:
I found out that `Key_{0-9}` can be converted to ints in [48,59] range.

Reply:
Alright, if that is reliable, you could make use of it. But am also fine with the current explicit version.